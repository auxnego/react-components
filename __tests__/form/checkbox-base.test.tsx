import React from "react";
import {create} from "react-test-renderer";
import { render,act } from "@testing-library/react";

import CheckboxBase from "../../form/checkbox-base";

it("Only id and value", () => {

  expect(create(
    <CheckboxBase id="example" value="ejemplo" />
  ).toJSON()).toMatchSnapshot();

});

it("with id, name, value and checked", () => {

  expect(create(
    <CheckboxBase id="ex2" name="example2" value="ejemplo2" checked />
  ).toJSON()).toMatchSnapshot();

});

it("Adding setProps for input and click event", () => {
  let { container } = render(<CheckboxBase 
                               id="example3" 
                               value="ejemplo3"
                               setProps = {{input:{role:"checkbox"}, root:{role:"box"}}}
                             />);
  act(() => {
    let target = container.querySelector('#example3') as HTMLInputElement;
    target.click();
  });
 
  expect(container).toMatchSnapshot();

});
