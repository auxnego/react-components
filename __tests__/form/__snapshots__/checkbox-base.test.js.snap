// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Adding setProps for input and click event 1`] = `
<div>
  <div
    class="checkbox-base__root checkbox-base__root--checked"
  >
    <input
      class="checkbox-base__input"
      id="example3"
      name="example3"
      type="checkbox"
      value="ejemplo3"
    />
    <i
      class="checkbox-base__icon fa fa-check"
    />
  </div>
</div>
`;

exports[`Only id and value 1`] = `
<div
  className="checkbox-base__root"
>
  <input
    className="checkbox-base__input"
    id="example"
    name="example"
    onClick={[Function]}
    type="checkbox"
    value="ejemplo"
  />
</div>
`;

exports[`with id, name, value and checked 1`] = `
<div
  className="checkbox-base__root checkbox-base__root--checked"
>
  <input
    checked={true}
    className="checkbox-base__input"
    id="ex2"
    name="example2"
    onClick={[Function]}
    type="checkbox"
    value="ejemplo2"
  />
  <i
    className="checkbox-base__icon fa fa-check"
  />
</div>
`;
